nx/lib/nx/defn/grad.ex

```diff
@@ -239,6 +239,8 @@ defmodule Nx.Defn.Grad do
   end
 
   defp update_grads(:optional, [_call, expr], _ans, gs, _to_grad_ids, grads) do
+    gs = List.wrap(gs)
```

Comment:
Could you add a test to the grads test file that regresses this fix?

Reply:
Also, given the call at line 222, probably we should have a guard for lists here and call with a [g] list there.

WDYT?

Reply:
Sure, happy to add a test.  

I played around with possible changes a bit.  The call on 222 is mostly used for versions of `update_grad` which expect a tensor such as with `op` being `:sum`, with `:optional` being unusual.  So we cannot change the call site 222 without doing a more extensive refactor.  Also, the `:optional` version is called from both branches of the case (of which 222 is one), so we can't change it to take a tensor.

Reply:
Right, then I think this solution makes sense :)